### DISCUSSION TITLE
Implicitly Casting VARCHARS

### DISCUSSION THREAD

**Discussion by RealNoArms** [duckdb/duckdb#13264] [Q&A]
I have a Polars DataFrame loaded with string values and some SQL from another system written for a database that implicitly casts the strings to numbers (`SELECT SUM(varchar_field1 * varchar_field2) AS double_field FROM pl_df`). DuckDB doesn't seem to handle this, or am I missing something?

**Comment by Tishj** [comment]
This sounds like https://github.com/duckdb/duckdb/pull/10115

But STRING_LITERAL types are reserved to internals currently, I could see this making sense in places like numpy object columns if we have to stringify the object because we couldn't find a proper mapping

But if the column of the Polars dataframe is defined as VARCHAR then I don't think mapping this to STRING_LITERAL by default makes a lot of sense
